apps/routes/lib/repo.ex

```diff
@@ -1,26 +1,23 @@
 defmodule Routes.Repo do
   @moduledoc "Repo for fetching Route resources and their associated data from the V3 API."
 
+  @behaviour Routes.RepoApi
+
   require Logger
   use RepoCache, ttl: :timer.hours(1)
 
   import Routes.Parser
 
   alias JsonApi
   alias Routes.{Route, Shape}
-  alias V3Api.{Routes, Shapes}
+  alias V3Api.{Shapes}
 
   @default_opts [include: "route_patterns"]
 
-  @doc """
-
-  Returns a list of all the routes
-
-  """
-  @spec all() :: [Route.t()]
+  @impl Routes.RepoApi
```

Comment:
Do you need both the `@behavior` and the `@impl` declarations?

Reply:
`@impl` tells the compiler which interface this is a function for. It'll help you out by failing if you forget any implementations.